cg/services/illumina/file_parsing/models.py

```diff
@@ -28,3 +29,41 @@ class DemuxMetrics(BaseModel):
     lane: int = Field(..., alias=DemuxMetricsColumnNames.LANE)
     sample_internal_id: str = Field(..., alias=DemuxMetricsColumnNames.SAMPLE_INTERNAL_ID)
     read_pair_count: int = Field(..., alias=DemuxMetricsColumnNames.READ_PAIR_COUNT)
+
+
+class DsmcEncryptionKey(BaseModel):
+    """Model representing the response from a PDC query."""
+
+    date: str
+    key_path: str
```

Comment:
It seems as if whenever this field is accessed we want it as a path rather than a string. Using the Annotated pattern we could write this as:
```suggestion
    key_path: Annotated[Path, BeforeValidator(validate_sequencing_path)]
```